### PR TITLE
expire backoff queues after they've delivered their messages

### PR DESCRIPTION
**Pull request by mattbennett**
A small enhancement that tidies up RabbitMQ dashboards by removing empty backoff queues once they've deadlettered their messages.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ env:
   - TOX_ENV=py34-lib
   - TOX_ENV=py34-examples
 script:
-  - tox -e $TOX_ENV
+  - tox -e $TOX_ENV -- pip install eventlet==0.20.0
```
Comment (kooba) [inline .travis.yml:17]: i think you want to use `0.21.0` it has few DNS bugs fixed we stumbled upon.

**Comment by kooba** [comment]
👍  Eventlet version for tox would probably not matter.

File: nameko_amqp_retry/backoff.py [hunk 2]
```diff
@@ -109,6 +114,10 @@ def republish(self, backoff_exc, message, target_queue):
             headers['backoff'] = expiration
             expiration_seconds = float(expiration) / 1000
 
+            # force redeclaration; the publisher will skip declaration if
+            # the entity has previously been declared by the same connection
+            maybe_declare(queue, conn)
```
Comment (timbu) [inline nameko_amqp_retry/backoff.py]: is there a danger here if the connection has gone?  (do we need a retry policy?)

File: .travis.yml (hunk 1 above)
Comment (mattbennett) [reply .travis.yml:17]: 0.20.1 is broken on several OS's unfortunately, including the build environment here -- https://github.com/eventlet/eventlet/issues/401

File: nameko_amqp_retry/backoff.py (hunk 2 above)
Comment (mattbennett) [reply nameko_amqp_retry/backoff.py]: I think this will create a new connection every time, but we should have a retry policy anyway.

**Review by kooba — APPROVED** [review]

**Review by timbu — APPROVED** [review]

**Pull request merged**